src/networking/address.rs

```diff
@@ -1,25 +1,20 @@
-/// An enum containing both ipv6 and ipv4 addresses
-
-#[derive(Debug)]
-pub enum IPVersion {
-  IPV4,
-  IPV6,
-}
+use std::net::Ipv4Addr;
 
 /// IPV4 address
-pub struct IPAddress {
-  pub address: String,
+#[derive(Debug, PartialEq)]
+pub struct Address {
+  pub address: Ipv4Addr,
   pub port: u16,
-  pub version: IPVersion,
 }
 
-impl std::fmt::Display for IPAddress {
+impl Address {
```

Comment:
If the implementation is empty you can leave it out

Reply:
it's not anymore :)